Managers/MapManager.gd

```diff
@@ -27,22 +27,25 @@ var current_map: MapBase
 ## - Width should not increase or decrease by more than 2 per floor (this makes it certain all rooms on the map are accessible [br]
 var map_width_array: Array[int]
 
+## Room events that we do not want to be consecutive
+const no_consecutive_room_event: Array[String] = ["shop", "heal"]
```

Comment:
maybe an array of GlobalEnum.EventType would prevent mistyping the name in a string ?

Reply:
I'm using strings here because I'm using the event name to know which event it is like this:
> `if current_room_event_name == "heal"`
> `if current_room_event_name in no_consecutive_room_event:`

So we could use the Class of the resource or GlobalEnum.EventType but I don't think we defined an GlobalEnum.EventType in the Event classes

Reply:
Ah yes you would need to have a variable on each event indeed, because the class name is always a string :thinking: 
Mmh let's keep it like that then